### PR TITLE
es6 field faculty: add marker value for missing value

### LINKED ISSUE #1075: {filters} Make it possible to filter on records with "missing" values
# Story

Make it possible to filter on records with "missing" values

- [ ] without a publication status
- [ ] records without faculty
- [ ] without files

## Success criteria

- [ ] The filter option stands on top of the list
- [ ] The label says:
  - [ ] No publication status
  - [ ] No affiliation
  - [ ] Without files
- [ ] When selected, all the records without a publication status, faculty or files are shown
- [ ] Librarians and researchers can both see this feature

## Screenshot

Example:
Screenshot 2023-09-08 at 12 01 31
Screenshot 2023-09-08 at 15 09 35

Discussions about this topic: https://github.com/ugent-library/biblio-backoffice/issues/874

### PR DESCRIPTION
**Pull request by nicolasfranck**
Fixes #1075 

What it does:

* adds marker value `n/a` ("not assigned") to the list of values in the elasticsearch field `faculty`. This value is now translated as `[missing]`.
* adds marker value `n/a` ("not assigned") to the list of values in the elasticsearch field `has_files` (new). Translations are now `with files` and `without files`
* adds marker value `n/a` ("not assigned") to the list of values in the elasticsearch field `vabb_type`. This value is now translated as `[missing]`.

TODO: reindex publications and datasets in backoffice; frontend not needed.

Remarks:
* there was also a possibility to search for empty values (a sort of "must not exists" query), but the way that was done added too much complexity (e.g. now we filter out facet filters if there the filter has the same name of the facet, and it is a term filter). Anyway, in the url we still had to come up with a special value to query on.
* the value `n/a` was so chosen so it could also be understood by end users. It appears by the way when you select the facet ("faculty: n/a") in the GUI

### CODE REVIEW

**Comment by nicolasfranck** [comment]
@nics @mietcls  will add support for missing year too? See #916

**Comment by nicolasfranck** [comment]
to delete or split up? @netsensei @nics

**Comment by netsensei** [comment]
@mietcls  I understand the question regarding missing field marker for faculties; but the other fields aren't mentioned in the issue #1075? It's unclear: is this PR finished, or do more fields need a "missing" option? Could you re-write / expand #1075 with the fields that need such a "missing" marker?

**Comment by mietcls** [comment]
@netsensei this issue is referenced as well, @nicolasfranck closed it as a double, but this asks for the other ones as well: https://github.com/ugent-library/biblio-backoffice/issues/915

**Review by nics — CHANGES_REQUESTED** [review]

File: internal/translations/translations.go [hunk 1]
```diff
@@ -304,6 +304,9 @@ func init() {
 	message.SetString(language.English, "builder.miscellaneous.publication", "Publication title")
 	message.SetString(language.English, "builder.miscellaneous.publication_abbreviation", "Publication short title")
 
+	message.SetString(language.English, "vabb_type.n/a", "[missing]")
+
+	message.SetString(language.English, "organization.n/a", "[missing]")
```
[the remark this answers is not in the document]
Comment (nicolasfranck) [reply internal/translations/translations.go:309]: @nics the title of the facet is "Faculty". A bit strange to put "no affiliation" below that, not?

[the remark this answers is not in the document]
Comment (nics) [reply internal/translations/translations.go:309]: yes but some organization.n/a could also be used for department, research group, ...

**Review by nics — APPROVED** [review]

**Pull request merged**